tests/integration/helpers.py

```diff
@@ -0,0 +1,140 @@
+
+    Returns: A tuple with the share info and auth info to mount the CephFS share.
+    """
+    client = Client()
+    if client.instances.exists("microceph"):
+        _logger.info("Microceph server already exists")
+        instance = client.instances.get("microceph")
+        return _get_cephfs_info(instance)
+
+    _logger.info("Bootstrapping Microceph cluster")
+    config = {
+        "name": "microceph",
+        "source": {
+            "alias": "jammy/amd64",
+            "mode": "pull",
+            "protocol": "simplestreams",
+            "server": "https://cloud-images.ubuntu.com/releases",
+            "type": "image",
+        },
+        "type": "virtual-machine",
+    }
+    client.instances.create(config, wait=True)
+    instance = client.instances.get(config["name"])
+    instance.start(wait=True)
+    _logger.info("Installing Microceph inside LXD container")
+
+    time = 2
+    for x in range(0, 10):
+        try:
+            _exec_command(instance, ["echo", "'Hello!'"])
+            err = None
+        except Exception as e:
+            err = e
+
+        if err:
+            _logger.info(f"Sleeping for {time} seconds")
+            sleep(time)
+            time *= 2
+        else:
+            break
+
+    if err:
+        raise err
+
+    _exec_commands(
+        instance,
+        [
+            ["ln", "-s", "/bin/true", "/usr/local/bin/udevadm"],
+            ["apt-get", "-y", "update"],
+            ["apt-get", "-y", "install", "ceph-common"],
+            ["snap", "install", "microceph"],
+            ["microceph", "cluster", "bootstrap"],
+            ["microceph", "disk", "add", "loop,2G,3"],
+            ["microceph.ceph", "osd", "pool", "create", f"{FS_NAME}_data"],
+            ["microceph.ceph", "osd", "pool", "create", f"{FS_NAME}_metadata"],
+            ["microceph.ceph", "fs", "new", FS_NAME, f"{FS_NAME}_metadata", f"{FS_NAME}_data"],
+            ["microceph.ceph", "fs", "authorize", FS_NAME, f"client.{USERNAME}", PATH, "rw"],
+            # Need to generate the test files inside microceph itself.
+            [
+                "ln",
+                "-sf",
+                "/var/snap/microceph/current/conf/ceph.client.admin.keyring",
+                "/etc/ceph/ceph.client.admin.keyring",
+            ],
+            [
+                "ln",
+                "-sf",
+                "/var/snap/microceph/current/conf/ceph.keyring",
+                "/etc/ceph/ceph.keyring",
+            ],
+            ["ln", "-sf", "/var/snap/microceph/current/conf/ceph.conf", "/etc/ceph/ceph.conf"],
+        ],
+    )
+
+    # Wait until the cluster is ready to mount the filesystem.
+    time = 2
+    for x in range(0, 10):
+        status = json.loads(_exec_command(instance, ["microceph.ceph", "-s", "-f", "json"]))
+        if status["health"]["status"] == "HEALTH_OK":
+            break
+        _logger.info(f"Sleeping for {time} seconds")
+        sleep(time)
+        time *= 2
```

Comment:
Same thing, we can use [tenacity](https://tenacity.readthedocs.io/en/latest/) here to continuously test if microceph is good to go.